lib/chef/provider/windows_task.rb

```diff
@@ -201,308 +224,303 @@ def action_disable
-            current_resource.cwd != new_resource.cwd ||
-            current_resource.frequency_modifier != new_resource.frequency_modifier ||
-            current_resource.frequency != new_resource.frequency ||
-            current_resource.idle_time != new_resource.idle_time ||
-            random_delay_updated? || execution_time_limit_updated? ||
-            (new_resource.start_day && new_resource.start_day != "N/A" && start_day_updated?) ||
-            (new_resource.start_time && new_resource.start_time != "N/A" && start_time_updated?)
-        begin
-          return true if new_resource.day.to_s.casecmp(current_resource.day.to_s) != 0 ||
-              new_resource.months.to_s.casecmp(current_resource.months.to_s) != 0
-        rescue
-          Chef::Log.debug "caught a raise in task_needs_update?"
+      def update_task(task)
+        converge_by("#{new_resource} task updated") do
+          task.application_name = new_resource.command
+          task.trigger = trigger
+          # TODO: Check update for execution time limit
+          task.max_run_time = new_resource.execution_time_limit.to_i * 1000
+          task.configure_settings(config_settings)
+          task.configure_principals(principal_settings)
         end
-
-        false
       end
 
-      # Comparing random_delay values using ISO8601::Duration object Ref: https://github.com/arnau/ISO8601/blob/master/lib/iso8601/duration.rb#L18-L23
-      # di = ISO8601::Duration.new(65707200)
-      # ds = ISO8601::Duration.new('P65707200S')
-      # dp = ISO8601::Duration.new('P2Y1MT2H')
-      # di == dp # => true
-      # di == ds # => true
-      def random_delay_updated?
-        if new_resource.random_delay.nil?
-          false
-        elsif current_resource.random_delay.nil? && new_resource.random_delay == "PT0S" # when user sets random_dealy to 0 sec
-          false
-        elsif current_resource.random_delay.nil?
-          true
-        else
-          ISO8601::Duration.new(current_resource.random_delay) != ISO8601::Duration.new(new_resource.random_delay)
+      def trigger
+        date = new_resource.start_day.split('/') if new_resource.start_day
+        time = new_resource.start_time.split(':') if new_resource.start_time
+        #TODO currently end_month, end_year and end_year needs to be set to 0. If not set win32-taskscheduler throwing nil into integer error.
+        trigger_hash = {}
+        trigger_hash = {
+          start_year: date.nil? ? 0 : date[2].to_i,
+          start_month: date.nil? ? 0 : date[0].to_i,
+          start_day: date.nil? ? 0 : date[1].to_i,
+          start_hour: time.nil? ? 0 : time[0].to_i,
+          start_minute: time.nil? ? 0 : time[1].to_i,
+          end_month: 0,
+          end_day: 0,
+          end_year: 0,
+          trigger_type: trigger_type,
+          type: type,
+          random_minutes_interval: new_resource.random_delay
+        }
+        if new_resource.frequency == :minute
```

Comment:
new_resource.frequency == :minute,
these conditions may directly be checked in convert_in_minutes method,
possible DRY up